### PR TITLE
Add VolunteerAbility, and allow volunteers to receive notifications

### PR DESCRIPTION
**Pull request by tiegz**
Fixes https://github.com/helpwithcovid/covid-volunteers/issues/103 

Instead of becoming owners, this just adds abilities to volunteers. Thoughts?

screencap

### Todo

* [x] actually send notifications to volunteers w/the ability
* [x] add tests
* [ ] is there a better way to do the CSS flash on save?
* [x] maybe add separate labels for each ability instead of just calling `titleize`
* [ ] add uniq constraint to volunteer_id/permission, and validation on permission

### CODE REVIEW

File: app/controllers/projects_controller.rb [hunk 1]
```diff
@@ -128,6 +128,19 @@ def toggle_volunteer
...
+
+    volunteer = @project.volunteers.find(update_volunteer_params[:id])
+    update_volunteer_params[:volunteer_abilities_attributes].each_pair do |idx, attrs|
+      volunteer.volunteer_abilities.find_or_initialize_by(permission: attrs[:permission]).tap do |ability|
+        ability.update(enabled: attrs["enabled"] == "1")
+      end
+    end
+
+    head :ok
+  end
```
Comment (tiegz) [inline app/controllers/projects_controller.rb:142]: Might be time to do a separate VolunteersController soon.

**Comment by jamiew** [comment]
Neat, so the idea is to break "owner" into discrete permissions?

* Receive volunteer notifications (current?)
* Edit project (?)
* Identify publicly as owner (?)

That kind of thing? That seems interesting to me, thanks for submitting!

Am I right that for now we'd just be using one abilities configuration for a "project owner"?

This could also be really useful for doing regional moderation

**Comment by tiegz** [comment]
> Neat, so the idea is to break "owner" into discrete permissions?

👍 Alternatively, it could be role-based and we could add a `Role` table instead, e.g. `@volunteer.roles << Role.find_by_name('email_assistant')`

> Am I right that for now we'd just be using one abilities configuration for a "project owner"?

Yeah, I was thinking of a simple  `<pseudocode>receive_notification if user_is_owner || user_has_the_right_ability</pseudocode>`

**Comment by tiegz** [comment]
Just noticed I haven't had time to shepherd this to the end. Gonna close, but feel free to reopen this if anyone's interested!

**Pull request closed without merging**